crates/noirc_evaluator/src/ssa/optim.rs

```diff
@@ -170,7 +170,7 @@ pub fn block_cse(
             let mut to_update_phi = false;
             anchor
                 .entry(ins.operator)
-                .or_insert_with(|| VecDeque::new());
+                .or_insert_with(|| {VecDeque::new()});
```

Comment:
The clippy error from before was suggesting `.or_insert_with(VecDeque::new)` - ie. no need to wrap it in a function, it already is one

Reply:
If I do not wrap it, clippy complains about redundant closure:
.or_insert_with(|| VecDeque::new());
    |                                 ^^^^^^^^^^^^^^^^^^ help: replace the closure with the function itself: `VecDeque::new`
If I follow this other suggestion from clippy I have now a compile error.
The best way I could find to get away from clippy was to warp it in a block.

Reply:
Thank you, I see what I was doing wrong: I was keeping the closure: `|| VecDeque::new `instead of just `VecDeque::new`